### DISCUSSION TITLE
Randozimation: Unequal block size

### DISCUSSION THREAD

**Discussion by xpressio** [rubenarslan/formr.org#577] [General]
Dear formr creators & fellow users,

I have noticed an issue with the buitlin 'shuffle' function: The randomization appears to favour one of my two conditions, leading to an increasing difference between the size of my two groups: currently, the total number of participants randomized is 180, with group A = 83 and group B = 97, difference 14. This is problematic, since we will soon have reached our preregistered sample size.

After looking into this a bit further, I noticed that up until August of this year, this difference was never more than 5 participants (data collection started in Feb 2022).

Has anyone else observed something similiar, do you have any idea what to do about this? Any help is greatly appreciated.

Best wishes,

Max

**Comment by rubenarslan** [comment]
Hi Max,

the p value of a binomial test for this difference is 0.33, so it's consistent with chance variation. Since I recall no changes to the shuffling code, I am not convinced you're seeing anything but regular randomization, which can sometimes cause larger differences in group sizes if you're unlucky. Unless you have other evidence that this is non-random, thats what I think is happening. 

If you don't want unequal group sizes, you can create your own randomization using calculate-type items. If you know the sample size is 120, you could do something like

```
set.seed(123)
groups <- sample(c(rep(0, 60), rep(1, 60)))
groups[iteration]
```

`iteration` is a variable at the survey level that starts at 1 and increments for each row. It was introduced in v0.21.4 and is only present in newly created surveys.
